libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -226,6 +226,28 @@ bool SyntaxChecker::visit(FunctionDefinition const& _function)
 				"\"."
 			);
 	}
+
+	if (v050)
+	{
+		// The parser still regards functions with the same name as the contract as constructors.
+		// In this case the function name will be non-emptpy, whereas it will be empty for
+		// constructors using the new syntax.
+		if (_function.isConstructor() && !_function.name().empty())
```

Comment:
Please add  a function to the function AST node called `isOldStyleConstructor` and use it here. In this case, the comment might not even be necessary.